html/README.md

```diff
@@ -0,0 +1,5 @@
+# HTML
+
+- Prefer input `type=text` paired with the [`inputmode`] attribute, to `type=number`.
```

Comment:
Why not both?

Reply:
See here https://buoy-software.slack.com/archives/C04GXBNK9BJ/p1699375450341469?thread_ts=1699375336.950699&cid=C04GXBNK9BJ

Reply:
Tried out both of these and they appear to give the same result, but `[type=number]` is still more semantic:
```html
  <label for="text-numeric">type=text + inputmode=numeric</label>
  <input type="text" inputmode="numeric" id="text-numeric" />
  <label for="numeric-number">type=number + inputmode=numeric</label>
  <input type="number" inputmode="numeric" id="numeric-number" />
```
https://github.com/BuoySoftware/guides/assets/711717/023fbffd-e7a7-4c66-8c4b-79ccc6c92a1e

Reply:
I seem to remember some browsers incremented decimal points, but that may be in the past now (capybara does, I remember having trouble while testing).

Tyson quotes accessibility issues and unwanted inc/decrement arrows.

Reply:
Right - `type=number` renders those undesired increment/decrement arrows, particularly on desktop browsers…

<img width="479" alt="Screenshot 2023-11-09 at 12 58 44" src="https://github.com/BuoySoftware/guides/assets/903327/a2b9a536-50df-41af-83ce-a933c053fa8c">

Generally speaking, those steppers are not useful to users and can often time clutter up the UI/get in the way.

As for semantics, `text` inputs are quite generic (I read it as any type of text, numbers included), so I think they still work in lots of places. In my opinion, the input’s label and any hints (if necessary) matter more and should be highly descriptive as to what type of value we want users to input. 

Screen readers will typically announce `type=number` inputs as ‘stepper’ or ‘increment’ inputs, potentially adding confusion.

This [great article gives a nice checklist for when a number input should probably be used](https://adamsilver.io/blog/form-design-when-to-use-the-number-input/):

> - incrementing and decrementing makes sense
> - the number doesn’t have a leading zero
> - the value doesn’t contain letters, slashes, minus signs, and decimal points.
> - the number isn’t very large

Perhaps we expand our guide addition here to include these details?

Reply:
Thanks @tysongach! I've updated to include a link to "Form design: when to use the number input"